src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php

```diff
@@ -49,4 +50,13 @@ public function postUpdate(LifecycleEventArgs $args): void
             $this->eventBus->dispatch(new CatalogPromotionUpdated($entity->getCatalogPromotion()->getCode()));
         }
     }
+
+    public function postRemove(LifecycleEventArgs $args): void
+    {
+        $entity = $args->getObject();
+
+        if ($entity instanceof CatalogPromotionActionInterface) {
+            $this->eventBus->dispatch(new CatalogPromotionConfigurationRemoved());
```

Comment:
Why didn't you use an existing event `CatalogPromotionUpdated`? Removing configuration is also an update of catalog promotion, and what is more, It looks like the behaviour in listeners is the same 🤔